src/storage/src/streaming_table/mod.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod mem_table;
+pub mod state_table;
+
+#[cfg(test)]
+pub mod test_streaming_table;
+
+use std::sync::Arc;
+
+use itertools::Itertools;
+use risingwave_common::array::column::Column;
+use risingwave_common::array::{DataChunk, Row};
+use risingwave_common::buffer::{Bitmap, BitmapBuilder};
+use risingwave_common::catalog::Schema;
+use risingwave_common::types::VIRTUAL_NODE_COUNT;
+
+use crate::error::StorageResult;
+use crate::streaming_table::state_table::DEFAULT_VNODE;
+
+/// Represents the distribution for a specific table instance.
+#[derive(Debug)]
+pub struct Distribution {
```

Comment:
Will the `Distribution` and `TableIter` in this file be used by the future batch_table? If yes, it is better to put it outside of `streming_table`. How about organizing the folder like:
```
├──storage
│  ├── table
│     ├── <common stuff> ...
│     ├── streaming
│       ├── ....
│     ├── batch
│       ├── ....
```

Reply:
Make sense to me.

Reply:
Let's apply the new folder after finishing refactoring of `streaming_table` and `batch_table`.